shell/platform/android/platform_view_android.cc

```diff
@@ -59,18 +54,56 @@ class PlatformMessageResponseAndroid : public blink::PlatformMessageResponse {
   ftl::WeakPtr<PlatformView> view_;
 };
 
-}  // namespace
+static std::unique_ptr<AndroidSurface> InitializePlatformSurfaceGL() {
+  const PlatformView::SurfaceConfig offscreen_config = {
+      .red_bits = 8,
+      .green_bits = 8,
+      .blue_bits = 8,
+      .alpha_bits = 8,
+      .depth_bits = 0,
+      .stencil_bits = 0,
+  };
+  auto surface = std::make_unique<AndroidSurfaceGL>(offscreen_config);
+  return surface->IsOffscreenContextValid() ? std::move(surface) : nullptr;
+}
+
+static std::unique_ptr<AndroidSurface> InitializePlatformSurfaceVulkan() {
+#if SHELL_ENABLE_VULKAN
+  auto surface = std::make_unique<AndroidSurfaceVulkan>();
+  return surface->IsValid() ? std::move(surface) : nullptr;
```

Comment:
The ternary [prevents copy elision](https://stackoverflow.com/questions/11914691/copy-elision-move-constructor-not-called-when-using-ternary-expression-in-retur).